contracts/governance/Vesting/fouryear/FourYearVestingLogic.sol

```diff
@@ -137,6 +137,41 @@ contract FourYearVestingLogic is IFourYearVesting, FourYearVestingStorage, Appro
 		emit DividendsCollected(msg.sender, _loanPoolToken, _receiver, _maxCheckpoints);
 	}
 
+	/**
+	 * @notice Sign transactions - only Token Owner.
+	 * @dev The setImpl and changeTokenOwner functions can only be
+	 * executed when both vesting owner and token owner have approved. This
+	 * function ascertains the approval of token owner.
+	 * */
+	function signTransaction() public onlyTokenOwner {
+		require(!signed[msg.sender], "already signed");
+		signed[msg.sender] = true;
+	}
+
+	/**
+	 * @notice Change token owner.
+	 * @dev Changes token owner. Must be
+	 * approved by both token owner and multisig(vesting owner).
+	 * @param _newTokenOwner Address of new token owner.
+	 * */
+	function changeTokenOwner(address _newTokenOwner) public onlyOwner {
+		require(_newTokenOwner != address(0), "invalid token owner address");
+		require(signed[tokenOwner], "must be signed by token owner");
```

Comment:
here it is required that the tokenowner approves first before the change was suggested. 

it should be the other way round. First, a new owner is suggested, then confirmed.

```
function changeTokenOwner(address _newTokenOwner) public onlyOwner{ 
    //contract owner suggests the ownership transfer, add eequirement checks
    newOwner = _newTokenOwner;
}

function approveOwnershipTransfer() public onlyTokenOwner{
     require(newTokenOwner != address(0));
     tokenOwner = newTokenOwner;
}
```

Reply:
Done